### PR TITLE
Fixing an issue with HolyGrail for slit dependent parameters

### PR DESCRIPTION
**Pull request by jhennawi**
HolyGrail was not allowing the sigdetect and rms_threshold to be slit dependent via lists into the parser. Now fixed.

### CODE REVIEW

File: pypeit/core/wavecal/autoid.py [hunk 1]
```diff
@@ -1398,15 +1397,15 @@ def __init__(self, spec, par = None, ok_mask=None, islinelist=False, outroot=Non
         #self._sigdetect = self._par['sigdetect']
         #self._lowest_nsig = self._par['lowest_nsig']
         # JFH I'm not convinced that the codea actually does anything except use the lowest nsig, but am not sure
-        self._sigdetect = self._par['sigdetect']
+        #self._sigdetect = self._par['sigdetect']
```
Comment (profxj) [inline pypeit/core/wavecal/autoid.py]: remove the commented lines?

**Review by profxj — CHANGES_REQUESTED** [review]
Nice stuff.  Please add a test to insure
this is now working as intended.  Thanks!

File: pypeit/core/wavecal/autoid.py [hunk 2]
```diff
@@ -2480,13 +2485,13 @@ def solve_slit(self, slit, psols, msols, tcent_ecent, nstore=1, nselw=3, nseld=3
...
                 continue
-            elif tfinal_dict['rms'] < self._rms_threshold:
+            elif tfinal_dict['rms'] < wvutils.parse_param(self._par, 'rms_threshold', slit):
                 # Has a better fit been identified (i.e. more lines ID)?
                 if len(tfinal_dict['pixel_fit']) > len(final_dict['pixel_fit']):
                     patt_dict, final_dict = copy.deepcopy(tpatt_dict), copy.deepcopy(tfinal_dict)
         return patt_dict, final_dict
 
-    def solve_patterns(self, bestlist, tcent_ecent):
+    def solve_patterns(self, slit, bestlist, tcent_ecent):
```
Comment (rcooke-ast) [inline pypeit/core/wavecal/autoid.py:2507]: docstring?

**Review by rcooke-ast — APPROVED** [review]
I have no problems if the tests check out, and a new test is added to make sure this doesn't break in the future. I also add a comment about docstrings (since Kyle mentioned this in the Slack very recently). I appreciate that I wrote this code originally, so it is my bad...

File: pypeit/core/wavecal/autoid.py (hunk 2 above)
Comment (jhennawi) [reply pypeit/core/wavecal/autoid.py:2507]: @rcooke-ast you wrote this routine;) I'm just fixing a bug.

Comment (rcooke-ast) [reply pypeit/core/wavecal/autoid.py:2507]: Here you go!!

        """
        Solve the patterns and find the best ID of each detected line.

        Parameters
        ----------
        slit: int
            The ID of the slit
        bestlist: list of ndarray
            A 5 element list, each containing a numpy.ndarray, with the following values required for each index:
              0: central wavelength of the pattern
              1: central dispersion of pattern
              2: sign of the pattern (note, sign = 1 [-1] if pixels correlate [anticorrelate] with wavelength
              3: index of the full list of patterns that were created from the detected arc lines
              4: index of the full list of patterns that were created from the line list.
        tcent_ecent : list
            A list [tcent, ecent] indicating which detection list
            should be used. Note that if arr_err is set then the weak
            keyword is ignored.

        Returns
        -------
        patt_dict : dict
            Dictionary containing information about the best patterns.
        """

File: pypeit/core/wavecal/autoid.py (hunk 1 above)
Comment (jhennawi) [reply pypeit/core/wavecal/autoid.py]: Done.

File: pypeit/core/wavecal/autoid.py (hunk 2 above)
Comment (jhennawi) [reply pypeit/core/wavecal/autoid.py:2507]: Added. Thanks!

**Comment by jhennawi** [comment]
Screenshot 2021-10-07 at 22 09 04

Unit tests pass.

**Comment by debora-pe** [comment]
Tests added in the Dev Suite https://github.com/pypeit/PypeIt-development-suite/pull/176
I'll run the entire Dev Suite over the weekend

**Comment by debora-pe** [comment]
Dev Suite passes.
```
--- PYPEIT DEVELOPMENT SUITE PASSED 129/129 TESTS (Masters ignored) ---
Testing Started at 2021-11-12T17:00:08.877161
Testing Completed at 2021-11-12T22:38:36.218152
Total Time: 5:38:27.340991
```

And unit tests too.

```
pytest . -W ignore
============================================================================================================= test session starts =============================================================================================================
platform darwin -- Python 3.8.5, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /Users/dpelliccia/PypeIt/PypeIt, configfile: setup.cfg
collected 302 items                                                                                                                                                                                                                           

test_alignments.py .                                                                                                                                                                                                                    [  0%]
test_arc.py .                                                                                                                                                                                                                           [  0%]
test_archive.py ..                                                                                                                                                                                                                      [  1%]
test_arcimage.py ..                                                                                                                                                                                                                     [  1%]
test_biasframe.py ..                                                                                                                                                                                                                    [  2%]
test_bitmask.py ....                                                                                                                                                                                                                    [  3%]
test_bpmimage.py ..                                                                                                                                                                                                                     [  4%]
test_bspline.py ...............                                                                                                                                                                                                         [  9%]
test_buildimage.py ..                                                                                                                                                                                                                   [ 10%]
test_calibrations.py .......                                                                                                                                                                                                            [ 12%]
test_chk_calibs.py ..                                                                                                                                                                                                                   [ 13%]
test_coadd.py .                                                                                                                                                                                                                         [ 13%]
test_collate_1d.py ............                                                                                                                                                                                                         [ 17%]
test_cooked.py .                                                                                                                                                                                                                        [ 17%]
test_coremeta.py .                                                                                                                                                                                                                      [ 18%]
test_datacontainer.py ........                                                                                                                                                                                                          [ 20%]
test_detector.py ...                                                                                                                                                                                                                    [ 21%]
test_display.py .                                                                                                                                                                                                                       [ 22%]
test_edgetrace.py .                                                                                                                                                                                                                     [ 22%]
test_fitting.py ....                                                                                                                                                                                                                    [ 23%]
test_flatfield.py .                                                                                                                                                                                                                     [ 24%]
test_flexure.py ..                                                                                                                                                                                                                      [ 24%]
test_flux.py ....                                                                                                                                                                                                                       [ 26%]
test_fluxspec.py .....                                                                                                                                                                                                                  [ 27%]
test_frametype.py .                                                                                                                                                                                                                     [ 28%]
test_history.py ....                                                                                                                                                                                                                    [ 29%]
test_load.py .                                                                                                                                                                                                                          [ 29%]
test_load_images.py ................                                                                                                                                                                                                    [ 35%]
test_maskdesign_matching.py ...                                                                                                                                                                                                         [ 36%]
test_masterframe.py .                                                                                                                                                                                                                   [ 36%]
test_match.py ....                                                                                                                                                                                                                      [ 37%]
test_metadata.py ......                                                                                                                                                                                                                 [ 39%]
test_moment.py ......                                                                                                                                                                                                                   [ 41%]
test_msgs.py ..                                                                                                                                                                                                                         [ 42%]
test_opticalmodel.py ....                                                                                                                                                                                                               [ 43%]
test_parse.py ...                                                                                                                                                                                                                       [ 44%]
test_pca.py ........                                                                                                                                                                                                                    [ 47%]
test_processimage.py ..                                                                                                                                                                                                                 [ 48%]
test_procimg.py .....                                                                                                                                                                                                                   [ 49%]
test_pydl.py .                                                                                                                                                                                                                          [ 50%]
test_pypeit.py ..                                                                                                                                                                                                                       [ 50%]
test_pypeitimage.py ..                                                                                                                                                                                                                  [ 51%]
test_pypeitpar.py ................................                                                                                                                                                                                      [ 61%]
test_pypeitsetup.py ........                                                                                                                                                                                                            [ 64%]
test_qa.py .                                                                                                                                                                                                                            [ 64%]
test_runpypeit.py ..                                                                                                                                                                                                                    [ 65%]
test_scienceimage.py ...                                                                                                                                                                                                                [ 66%]
test_scripts.py .................                                                                                                                                                                                                       [ 72%]
test_setups.py .....................                                                                                                                                                                                                    [ 79%]
test_skysub.py ..                                                                                                                                                                                                                       [ 79%]
test_slitmask.py ...                                                                                                                                                                                                                    [ 80%]
test_slittrace.py ....                                                                                                                                                                                                                  [ 82%]
test_spec2dobj.py ......                                                                                                                                                                                                                [ 84%]
test_specobj.py .......                                                                                                                                                                                                                 [ 86%]
test_specobjs.py .....                                                                                                                                                                                                                  [ 88%]
test_spectrographs.py ....................                                                                                                                                                                                              [ 94%]
test_syncspec.py .                                                                                                                                                                                                                      [ 95%]
test_telluric.py ..                                                                                                                                                                                                                     [ 95%]
test_traceimage.py .                                                                                                                                                                                                                    [ 96%]
test_utils.py ...                                                                                                                                                                                                                       [ 97%]
test_vcorr.py ..                                                                                                                                                                                                                        [ 97%]
test_wavemodel.py .                                                                                                                                                                                                                     [ 98%]
test_wavetilts.py ....                                                                                                                                                                                                                  [ 99%]
test_wvcalib.py ..                                                                                                                                                                                                                      [100%]

====================================================================================================== 302 passed in 1139.16s (0:18:59) =======================================================================================================

```

**Review by profxj — APPROVED** [review]
thx all!

**Pull request merged**